src/main/java/hu/bme/mit/spaceship/TorpedoStore.java

```diff
@@ -30,18 +31,17 @@ public TorpedoStore(int numberOfTorpedos){
 
   public boolean fire(int numberOfTorpedos){
     if(numberOfTorpedos < 1 || numberOfTorpedos > this.torpedoCount){
-      new IllegalArgumentException("numberOfTorpedos");
+      throw new IllegalArgumentException("numberOfTorpedos");
     }
 
     boolean success = false;
 
     // simulate random overheating of the launcher bay which prevents firing
-    Random generator = new Random();
-    double r = generator.nextDouble();
+    double r = rand.nextDouble();
```

Comment:
I would create a new instance of Random() on every call.
Something like:
`
new Random().nextDouble();
`